test/source-sink-errors/mzcompose.py

```diff
@@ -538,7 +539,8 @@ def workflow_default(c: Composition) -> None:
     introducing a Disruption and then checking the mz_internal.mz_*_statuses tables
     """
 
-    for disruption in disruptions:
+    sharded_disruptions = buildkite.shard_list(disruptions, lambda s: s.name)
```

Comment:
You might want to add something like
```
    print(
        f"Checks in shard with index {buildkite.get_parallelism_index()}: {[s.__name__ for s in scenarios]}"
    )
```

(We can also generalize that and move it into `shard_list`.)